src/thunderbots/software/ai/hl/stp/evaluation/enemy_threat.cpp

```diff
@@ -0,0 +1,169 @@
+#include "ai/hl/stp/evaluation/enemy_threat.h"
+
+#include <deque>
+
+#include "ai/hl/stp/evaluation/intercept.h"
+#include "ai/hl/stp/evaluation/possession.h"
+#include "ai/hl/stp/evaluation/team.h"
+#include "ai/world/world.h"
+#include "geom/util.h"
+#include "shared/constants.h"
+
+std::optional<std::pair<int, std::optional<Robot>>> Evaluation::getNumPassesToRobot(
+    const Robot& initial_passer, const Robot& final_receiver, const Team& team,
+    const std::vector<Robot>& other_robots)
+{
+    if (initial_passer == final_receiver)
+    {
+        return std::make_pair(0, std::nullopt);
+    }
+
+    // We calculate the minimum number of passes it would take for the initial_passer
+    // robot to pass the ball to the final_receiver, assuming both robots are on the given
+    // team
+    //
+    // This algorithm essentially treats the team of robots like a Directed Acyclic
+    // Graph and "traverses" the graph to find the shortest path to the robot
+
+    // The robots that could have the ball and make a pass at each iteration. They
+    // are on the "frontier" of the graph search
+    std::vector<Robot> current_passers{initial_passer};
+    // The remaining robots we haven't checked yet
+    std::vector<Robot> unvisited_robots = team.getAllRobots();
+    // Remove the robot already in the current_passers since it starts off as visited
```

Comment:
I'm a bit confused about what this means. Are we just removing the `initial_passer`?

Reply:
Yes that's all we're doing, since the algorithm already starts by "visiting" the initial passer. I've made this a bit more clear in the comments.